src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs

```diff
@@ -35,74 +38,105 @@ public class ContentService : IContentService
             _methodologyService = methodologyService;
         }
 
-        public async Task UpdatePublicationTree()
+        /**
+         * Intended to be used as a Development / BAU Function to perform a full content refresh
+         */
+        public async Task UpdateAllContentAsync()
         {
-            var tree = _publicationService.GetPublicationsTree();
-            await _fileStorageService.UploadFromStreamAsync(PublicContentPublicationsTreePath(), "application/json",
-                SerializeObject(tree, _jsonSerializerSettingsCamelCase));
+            // Content is not staged when performing a full refresh
+            const bool staging = false;
+
+            await _fileStorageService.DeleteAllContentAsync();
+            await UpdateTreesAsync(staging);
+
+            var publications = _publicationService.ListPublicationsWithPublishedReleases().ToList();
+            var methodologyIds = publications.Where(publication => publication.MethodologyId.HasValue)
+                .Select(publication => publication.MethodologyId.Value).Distinct();
+
+            foreach (var publication in publications)
```

Comment:
Just as an aside, you can also write as `publications.ForEach(publication => ...)` if you wanted to

Reply:
Thanks, I did write it this originally but it gave me the following exception. Do you have any ideas how to solve it @duncan-at-hiveit?

`System.InvalidOperationException: A second operation started on this context before a previous operation completed. This is usually caused by different threads using the same instance of DbContext. For more information on how to avoid threading issues with DbContext, see https://go.microsoft.com/fwlink/?linkid=2097913.
   at Microsoft.EntityFrameworkCore.Internal.ConcurrencyDetector.EnterCriticalSection()
   at Microsoft.EntityFrameworkCore.Query.RelationalShapedQueryCompilingExpressionVisitor.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at lambda_method(Closure , QueryContext )
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
   at System.Linq.Queryable.Last[TSource](IQueryable`1 source, Expression`1 predicate)
   at GovUk.Education.ExploreEducationStatistics.Publisher.Services.ReleaseService.GetLatestRelease(Guid id) in C:\Users\ben\dev\explore-education-statistics\src\GovUk.Education.ExploreEducationStatistics.Publisher\Services\ReleaseService.cs:line 69
   at GovUk.Education.ExploreEducationStatistics.Publisher.Services.ContentService.UpdateLatestReleaseAsync(Publication publication, Boolean staging) in C:\Users\ben\dev\explore-education-statistics\src\GovUk.Education.ExploreEducationStatistics.Publisher\Services\ContentService.cs:line 109
   at GovUk.Education.ExploreEducationStatistics.Publisher.Services.ContentService.<UpdateAllContentAsync>b__8_2(Publication publication) in C:\Users\ben\dev\explore-education-statistics\src\GovUk.Education.ExploreEducationStatistics.Publisher\Services\ContentService.cs:line 60
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()`

Reply:
Looks like by default it's returned you a parallelised Collection of some sort.  You could probably do a ToList() on it first to make it sequential, but tis no bother to be honest.  Interesting though!